frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/ModelManager.java

```diff
@@ -83,7 +83,7 @@ public ModelArchive registerModel(
 
         ModelArchive archive = ModelArchive.downloadModel(configManager.getModelStore(), url);
         if (modelName == null || modelName.isEmpty()) {
-            modelName = archive.getModelName();
+            modelName = archive.getModelName().isEmpty() ? url : archive.getModelName();
```

Comment:
1. This should be an error case, we should not use url ad model name.
2. archive.getModelName() might return null.

Reply:
I removed this check. I think the change above makes sure that the modelName is always set.